### PR TITLE
[dattri.func] Fix `ChunkedCudaProjector` bug

### PR DESCRIPTION
**Pull request by tingwl0122**
### 1. Motivation and Context
We found that `ChunkedCudaProjector` not working as expected for large models. For Dict input, we found it hard to project without dividing gradients, so we will inherently force dict input to be tensor and split the tensor for projection if it is too large.

### 2. Summary of the change
- fix chunkcudaprojector bug by forcing tensor input
- add chunkcudaprojector test case for tensor input

Note that I preserve most of the original code and possibly we can come back and make it support dict input (without enforcing it to be tensor).

### 3. What tests have been added/updated for the change?
- [x] Unit test: Typically, this should be included if you implemented a new function/fixed a bug.

### CODE REVIEW

File: dattri/func/projection.py [hunk 1]
```diff
@@ -457,7 +457,7 @@ def free_memory(self) -> None:
         del self.ch_input
         self.input_allocated = False
 
-    def project(self, features: Union[dict, Tensor], ensemble_id: int) -> Tensor:
+    def dict_project(self, features: Union[dict, Tensor], ensemble_id: int) -> Tensor:
```
Comment (TheaperDeng) [inline dattri/func/projection.py:460]: will this `dict_project` be used later?

File: dattri/func/utils.py [hunk 2]
```diff
@@ -88,7 +88,7 @@ def _vectorize(
     return arr
 
 
-def get_parameter_chunk_sizes(
+def _get_parameter_chunk_sizes(
```
Comment (TheaperDeng) [inline dattri/func/utils.py:91]: will this function be used later?

File: dattri/func/projection.py (hunk 1 above)
Comment (tingwl0122) [reply dattri/func/projection.py:460]: yeah, probably. If we are going to support dict input

File: dattri/func/utils.py (hunk 2 above)
Comment (tingwl0122) [reply dattri/func/utils.py:91]: as above, might be used together

**Review by TheaperDeng — APPROVED** [review]
LGTM, please also have a look @sx-liu

**Review by sx-liu — APPROVED** [review]
Also LGTM, works for MT5 model.

**Pull request merged**